src/State/ProcessorInterface.php

```diff
@@ -25,8 +25,9 @@
 interface ProcessorInterface
 {
     /**
-     * Handle the state.
+     * Handles the state.
      *
+     * @param T                                                                                                                                                      $data
```

Comment:
`$data` and the return type may be different can't we have another generic then `T`?